### PR TITLE
initial guardduty support

### PR DESCRIPTION
**Pull request by KingBrewer**
GuardDuty Support

### CODE REVIEW

File: src/erlcloud_guardduty.erl [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+
+%%%------------------------------------------------------------------------------
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+
+filter_undef(List) ->
+    lists:filter(fun({_Name, Value}) -> Value =/= undefined end, List).
+
+base_path() ->
```
Comment (motobob) [inline src/erlcloud_guardduty.erl]: ?do we really need this?

File: src/erlcloud_guardduty.erl [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                            {"MaxItems", MaxItems}]),
+    guardduty_request(Config, get, Path, undefined, QParams).
+
+
+%%%------------------------------------------------------------------------------
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+
+filter_undef(List) ->
+    lists:filter(fun({_Name, Value}) -> Value =/= undefined end, List).
```
Comment (motobob) [inline src/erlcloud_guardduty.erl]: i thought there was a common one for this

Comment (KingBrewer) [reply src/erlcloud_guardduty.erl]: Nope. The one you are referring to has apparently been declared within `erlcloud_lambda` module. I'll move it to the `erlcloud_util` one and import it from both modules respectively.

File: src/erlcloud_guardduty.erl (hunk 1 above)
Comment (KingBrewer) [reply src/erlcloud_guardduty.erl]: Of course we don't. I'm fixing it.

**Comment by KingBrewer** [comment]
@motobob I addressed your comments

**Pull request merged**